lib/structures/Channel.js

```diff
@@ -19,6 +19,33 @@ class Channel extends Base {
         return `<#${this.id}>`;
     }
 
+    static from(data, client) {
+        switch(data.type) {
+            case 0: {
+                return new TextChannel(data, client.guilds.get(data.guild_id));
```

Comment:
Also nitpick, but passing client and fetching the guild in GuildChannel would remove the need to do this every time